### ISSUE TITLE
"Exceeded max line length" errors on exceptionally long function names

### ISSUE THREAD

**Issue by bethcutler** [pinterest/ktlint#519]
Too-long line errors are triggered if the function name is exceptionally long.  This is more likely to occur in tests, e.g.

```
fun testExceptionallyLongFunctionName_exceptionallyLongScenario_exceptionallyLongExpectedOutcome() {
...
}
```

In most cases, I think it is preferable to rewrite the function name to be shorter, but arguably there are cases where having a function name that is longer than 100 characters is unavoidable.

**Comment by Tapchicoma** [comment]
You can always suppress error for this line with function name: https://github.com/pinterest/ktlint#how-do-i-suppress-an-error

Imho, this should be enough for such cases and as you've said, generally, function names should not be so long. I can agree, that sometimes test function name maybe long, but, personally, I prefer using [spek](https://spekframework.org/) or [kotlintest](https://github.com/kotlintest/kotlintest) frameworks to run and describe tests.

**Comment by shashachu** [comment]
Agreed. Since this is an exceptional case, it seems like the right candidate for suppression.

**Comment by bethcutler** [comment]
Thanks, I did not know about suppressing errors for individual lines!  I'll pass that along.

**Comment by shashachu** [comment]
Great. Let us know if you run into any issues.
